src/test/java/io/jenkins/plugins/file_parameters/FileParameterWrapperTest.java

```diff
@@ -58,4 +62,129 @@
         r.assertLogContains("loaded 'UPLOADED CONTENT HERE' from ", b);
     }
 
+    @Ignore("need to implement option to tolerate undefined parameter")
+    @Test public void base64Undefined() throws Exception {
+        r.createSlave("remote", null, null);
+        WorkflowJob p = r.createProject(WorkflowJob.class, "myjob");
+        p.addProperty(new ParametersDefinitionProperty(new Base64FileParameterDefinition("FILE", null)));
+        String pipeline = "pipeline {\n" +
+            "  agent any\n" +
+            "  parameters {\n" +
+            "    base64File(name: 'FILE')\n" +
+            "  }\n" +
+            "  stages {\n" +
+            "    stage('Example') {\n" +
+            "      steps {\n" +
+            "        withFileParameter('FILE') {\n" +
+            "          echo('foo') \n" +
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+        p.setDefinition(new CpsFlowDefinition(pipeline, true));
+        WorkflowRun run = p.scheduleBuild2(0).get();
+        r.waitForCompletion(run);
+        // definitely will fail but we just ensure parameter has been created
+        r.assertBuildStatus(Result.SUCCESS, run);
+        WorkflowRun b = p.getBuildByNumber(1);
+        r.assertLogContains("foo", b);
+    }
+
+    @Test public void base64DeclarativeParameterCreated() throws Exception {
+        r.createSlave("remote", null, null);
+        WorkflowJob p = r.createProject(WorkflowJob.class, "myjob");
+
+        String pipeline = "pipeline {\n" +
+            "  agent any\n" +
+            "  parameters {\n" +
+            "    base64File(name: 'FILE')\n" +
+            "  }\n" +
+            "  stages {\n" +
+            "    stage('Example') {\n" +
+            "      steps {\n" +
+            "        withFileParameter('FILE') {\n" +
+            "          echo(/loaded '${readFile(FILE).toUpperCase(Locale.ROOT)}' from $FILE/) \n" +
+            "        }\n" +
+            "      }\n" +
+            "    }\n" +
+            "  }\n" +
+            "}";
+
+        p.setDefinition(new CpsFlowDefinition(pipeline, true));
+        WorkflowRun run = p.scheduleBuild2(0).get();
+        r.waitForCompletion(run);
+        // definitely will fail but we just ensure parameter has been created
```

Comment:
You could use `[WorkflowJob.]addProperty` to define params before running the build. Or the build may be able to check at runtime

```groovy
env.FILE != null
```

(For Base64 params. Not for stashed params, which never set an env var at the build level.)

Reply:
the idea here was test the auto creation of the parameter after the first run